src/lib/match.ts

```diff
@@ -1,9 +1,13 @@
 import { assert } from 'sinon';
 import { getBaseTypeDiff } from './get-base-type-diff';
+import { expect } from 'chai';
 
 export function match(actual: unknown, expected: unknown) {
   let matchResult: boolean;
   try {
+    if (typeof actual === 'string' && typeof expected === 'string') {
```

Comment:
That works, good find!
I think that maybe other types may have the same treatment and the expect function may not be needed to use. Could you do something like this:
```ts
    if (actual !== expected) {
      if (shouldFailOnExactCompare(typeof actual, typeof expected)) {
        matchResult = false;
      } else {
        assert.match(actual, expected);
      }
    }	
```

And, in the implementation of **shouldFailOnExactCompare** you could just check if both types are string

Reply:
you are right!! I will make this change

Reply:
Done!